building/tracks/ci/gha-best-practices.md

```diff
@@ -0,0 +1,139 @@
+
+```yml
+permissions:
+  contents: read
+```
+
+Check the [GitHub Docs](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#permissions) for the full list of permissions.
+
+### Pin actions to SHAs
+
+### Consider setting up a concurrency strategy
+
+It's often not necessary or useful to run CI on intermediate commits if a newer commit has been pushed in the meantime.
+
+You can configure a [concurrency strategy](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#concurrency) in order to automatically cancel running workflows in the same context.
+
+#### Example
+
+To cancel intermediate builds in a PR, you can use the following concurrency settings:
+
+```yml
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+```
+
+<details><summary>Third Party Notice</summary>
+
+The example above is based on [PkgTemplates.jl's CI workflow](https://github.com/invenia/PkgTemplates.jl/blob/cb02b3416b1adcac8006dd4905947e0f2d6994aa/.github/workflows/CI.yml#L10-L14), published under the [MIT license](https://github.com/invenia/PkgTemplates.jl/blob/cb02b3416b1adcac8006dd4905947e0f2d6994aa/LICENSE):
+
+> MIT License
+> 
+> Copyright (c) 2017-2020 Chris de Graaf, Invenia Technical Computing Corporation
+> 
+> Permission is hereby granted, free of charge, to any person obtaining a copy
+> of this software and associated documentation files (the "Software"), to deal
+> in the Software without restriction, including without limitation the rights
+> to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+> copies of the Software, and to permit persons to whom the Software is
+> furnished to do so, subject to the following conditions:
+> 
+> The above copyright notice and this permission notice shall be included in all
+> copies or substantial portions of the Software.
+> 
+> THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+> IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+> FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+> AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+> LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+> OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+> SOFTWARE.
+
+
+</details>
+
+### Consider which triggers are really needed
+
+### Read the "Security hardening for GitHub Actions" guide
+
+The practices mentioned above are by no means exhaustive.
+For a comprehensive guide on good security practices for using GitHub Actions safely, check out [GitHub's security guide](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions).
```

Comment:
Is this part of the `Read the "Security hardening guide...` section or its own section? I assume the latter, I. Which case it may be better placed under the. `Collection of Best Practices` heading where it may appear more separated?

Reply:
This is the `Read the "Security hardening guide...` section, I wanted to add some explanation why one should read it instead of only linking to it.

Reply:
Do you think it should be one level higher in the hierarchy?